learning_paths/api/v1/views.py

```diff
@@ -57,3 +64,56 @@ def get(self, request, learning_path_uuid):
         if serializer.is_valid():
             return Response(serializer.data, status=status.HTTP_200_OK)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class LearningPathUserGradeView(APIView):
+    """
+    API view to return the aggregate grade of a user in a learning path.
+    Only accessible by staff users.
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, learning_path_uuid):
+        """
+        Fetch learning path grade
+        """
+        if not request.user.is_staff:
+            return Response(
+                {"detail": "Permission denied."}, status=status.HTTP_403_FORBIDDEN
+            )
+
+        username = request.query_params.get("username")
+        if not username:
+            return Response(
+                {"detail": "Username parameter is required."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        user = get_object_or_404(User, username=username)
+
+        learning_path = get_object_or_404(LearningPath, uuid=learning_path_uuid)
+
+        try:
+            grading_criteria = LearningPathGradingCriteria.objects.get(
+                learning_path=learning_path
+            )
+        except LearningPathGradingCriteria.DoesNotExist:
+            return Response(
+                {"detail": "Grading criteria not found for this learning path."},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        is_completed = is_learning_path_completed(user, learning_path)
+        aggregate_grade = calculate_learning_path_grade(user, learning_path)
+        meets_expected_grade = aggregate_grade >= grading_criteria.expected_grade
```

Comment:
We have a dedication function (or a method, depending on when you read this) that returns `aggregate_progress >= grading_criteria.completion_threshold`, so we should do the same thing here for consistency.

Reply:
That one was for completion threshold. I wanted to have one for grade threshold too. Updated the variable names so it should be more clear now.